### DISCUSSION TITLE
[Enterprise Use] Support for one-way file transfer and clipboard transfer

### DISCUSSION THREAD

**Discussion by emc2-2022** [rustdesk/rustdesk#7837] [Q&A] [answered]
[Enterprise Use] Support for one-way file transfer and clipboard transfer

**Background**: In enterprise applications, employees participate in confidential project development. One-way transfer can meet the requirements of security and convenience, making it a practical feature.

**Requirement**: Can the policy page customize one-way clipboard and file transfer?

**Purpose**: Prohibit the controlled end from transferring content to the controlling end, and allow the controlling end to transfer content to the controlled end.

> from a rustdesk pro user.

**Comment by rustdesk** [comment]
Follow this up after finishing android mediacodec. The one way means only from local to remote, no need to consider from remote to local  for the time being.

- [x] also make sure work for file manager
- [x] BTW, add file/dir rename functionality to file manager, https://github.com/rustdesk/rustdesk/discussions/7758
- [x] Better tab bar. 1, dynamic tab width (reference chrome) 2, A droplist to choose tab when no space to show all (chrome does not have this droplist, but it will shrink all tabs together, seems not good). https://github.com/rustdesk/rustdesk/pull/7918
- [ ] reorderable tab


![image](https://github.com/rustdesk/rustdesk/assets/71636191/bc65032b-95d1-4e20-8159-c02e806da958)

![image](https://github.com/rustdesk/rustdesk/assets/71636191/2100d690-c71a-4bd3-a44c-4e4477e46dc5)

**Reply by emc2-2022** [reply]
Thank you, this will be a very useful feature, and we are looking forward to it!

**Comment by rustdesk** [comment, marked as answer]
https://rustdesk.com/docs/en/self-host/client-configuration/advanced-settings/#one-way-clipboard-redirection

https://rustdesk.com/docs/en/self-host/client-configuration/advanced-settings/#one-way-file-transfer

**Comment by Anuril** [comment]
We think the opposite direction should also be controllable, we want the clipboard of the controlled computer to be transferred to the controlling side, but not the clipboard of the controlling side to the client computer. If we use [disable-clipboard](https://rustdesk.com/docs/en/self-host/client-configuration/advanced-settings/#disable-clipboard), we loose the functionality in both directions.

**Reply by aidenpadv** [reply]
Seconded. For our use, the ability to copy event log text from a remote computer without having to worry about passwords copied from our own computers showing up in the remote computer's clipboard, would be extremely helpful.
